src/App.tsx

```diff
@@ -17,18 +98,34 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={handleStatusChange}
+                onSearchChange={handleSearchChange}
+                onClearSearch={handleClearSearch}
+                searchQuery={searchQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+
+              <TodoList
+                todos={getFilteredTodos()}
+                onSelectTodo={handleSelectTodo}
+                selectedTodoId={selectedTodo?.id}
+              />
```

Comment:
Ensure that the `TodoList` component correctly accepts and types its props. This includes `todos`, `onSelectTodo`, and `selectedTodoId`.